### ISSUE TITLE
`semicolon_if_nothing_returned` on an attribute macro

### ISSUE THREAD

**Issue by Gingeh** [rust-lang/rust-clippy#12123]
### Summary

Clippy suggests to add a semicolon after the `pollster::main` attribute macro. Interestingly, it doesn't trigger when the function is written on a single line.

### Lint Name

semicolon_if_nothing_returned

### Reproducer

I tried this code:

```rust
#[pollster::main]
async fn main() {
}
```

I saw this happen:

```
warning: consider adding a `;` to the last statement for consistent formatting
 --> src/main.rs:1:1
  |
1 | #[pollster::main]
  | ^^^^^^^^^^^^^^^^^ help: add a `;` here: `#[pollster::main];`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#semicolon_if_nothing_returned
  = note: `-W clippy::semicolon-if-nothing-returned` implied by `-W clippy::pedantic`
  = help: to override `-W clippy::pedantic` add `#[allow(clippy::semicolon_if_nothing_returned)]`
```

### Version

```text
rustc 1.77.0-nightly (75c68cfd2 2024-01-07)
binary: rustc
commit-hash: 75c68cfd2b9870f2953b62d250bd7d0564a7b56d
commit-date: 2024-01-07
host: x86_64-unknown-linux-gnu
release: 1.77.0-nightly
LLVM version: 17.0.6
```

### Additional Labels

* `I-suggestion-causes-error`
* `T-macros`
